packages/store-sync/src/postgres-decoded/buildTable.ts

```diff
@@ -46,14 +47,21 @@ export function buildTable<TKeySchema extends KeySchema, TValueSchema extends Va
   keySchema,
   valueSchema,
 }: BuildTableOptions<TKeySchema, TValueSchema>): BuildTableResult<TKeySchema, TValueSchema> {
-  const schemaName = transformSchemaName(`${getAddress(address)}__${namespace}`);
+  // We snake case the schema name and table name for ergonomics in querying
+  // (no need to double-quote), but this is destructive and may lead to
+  // overlapping namespaces (both `MyNamespace` and `my_namespace` snake case
+  // to the same value).
+  // TODO: consider not snake casing these
```

Comment:
@alvrs Curious if you have thoughts on this!

This snake case approach was inspired by Sky Strife analytics queries, where the SQL is quite ugly due to all the double quotes for case sensitivity:
https://github.com/latticexyz/mud/compare/historical-indexer?expand=1#diff-ec8025f1e682cd379feaad115d8f46b1220fc4a37e1859dc9a360e8b54fd9d14

Reply:
If we wanted to only snake case column names, the best we can do with SQL for case sensitive schemas/tables is something like

```sql
SET search_path TO "0x7203e7ADfDF38519e1ff4f8Da7DCdC969371f377__";

SELECT
    __last_updated_block_number AS created_at_block,
    created_by,
    encode(level_id, 'escape') AS map,
    key AS match_entity
FROM
    "MatchConfig" match_config;
```

- set `search_path` to the schema(s) we want to query from with their "shorthand" form (i.e. just table name)
- manually alias case-sensitive `MatchConfig` table name to `match_config`, in case we want to do joins without double quotes
- column names can be safely snake cased

For ORMs and other programmatic SQL tools, I'm not sure how easy it is to use something like `search_path`, but maybe these would also abstract over the quoting of things like schemas to ensure case sensitivity.